backend/monitoring_integrations/newrelic_adapter.py

```
@@ -0,0 +1,36 @@
+                })
+            elif "violations" in payload and isinstance(payload["violations"], list):
+                for v in payload["violations"]:
+                    alerts.append({
+                        "provider": "newrelic",
+                        "component": v.get("condition_name", "newrelic_violation"),
+                        "severity": v.get("severity", "warning"),
+                        "message": v.get("description", ""),
+                        "metadata": v,
+                    })
+            else:
+                raise AdapterParseError("Unrecognized New Relic payload")
```

Ruff (0.14.10)

Abstract `raise` to an inner function

(TRY301)